Added development process (#66)
SW709E19/Report: report/Appendix/DevelopmentProcess.tex

\section{Development Process}

> To manage the development process of this project we have been using Scrum. \cite{ScrumGuide} Scrum is a framework we have learned about through earlier courses and is a framework which is used heavily within the industry.

Reviewer: ```suggestion
To manage the development process of this project we have been using Scrum. \cite{ScrumGuide} 
Scrum is a framework we have learned about through earlier courses and is a framework which is used heavily within the industry.
```

> The Scrum framework uses teams with a size between 3-9 members, so since there are 6 members in our group, we will only have a single team. Within a Scrum team, there are different roles. These are: The Scrum master, the product owner and the development team. In our case, Thomas Rusbjerg from our group is the Scrum master, Rasmus Steiniche from the company Neurospace is the product owner and the rest of the members are the development team.

Reviewer: ```suggestion
The Scrum framework uses teams with a size between 3-9 members, so since there are 6 members in our group, we will only have a single team. Within a Scrum team, there are different roles. These are The Scrum Master, the product owner and the development team. In our case, Thomas Rusbjerg from our group is the Scrum master, Rasmus Steiniche from the company Neurospace is the product owner and the rest of the members are the development team.
```

Reviewer: Der skal kilder på tal omkring antal medlemmer 3-9

Author: Hvad er argumentet for ingen : og stort M. Grammarly foreslog det også, men virker forkert?

Den ene kilde der er bruges til det hele, men jeg kan godt cite den igen, hvis det er bedre sådan.

Author: Så er der kilde på.

> The Scrum master is responsible for ensuring that the Scrum framework is used correctly, and to mentor others within the group about the Scrum framework.

Reviewer: ```suggestion
The Scrum Master is responsible for ensuring that the Scrum framework is used correctly, and to mentor others within the group about the Scrum framework.
```

The product owner is responsible for maximising the value of the product. The product owner is the one that manages the product backlog.

> The task for the development team is to deliver increments of "done" at the end of every sprint.

Reviewer: Måske en defination af hvad "done" er

The Scrum framework contains several events that are useful for the development process. These are: The sprint, sprint planning, sprint goal, sprint review, sprint retrospective and the daily Scrum.

The sprint is a duration of time usually less than a month and in our case just 2 weeks. During a sprint, no changes can be made that would alter the sprint goal. Within this time frame, an amount of work is "done" which should result in a new increment of the product.

> Sprint planning is done at the start of every sprint. The plan is created by the entire Scrum team working together. It should take at most 8 hours for 1-month sprints, so in our case, it should at most take 4 hours since our sprint duration is 2 weeks. During the sprint planning, we show the progress since the last sprint planning to the product owner, after which the product backlog gets updated with new requirements that build upon the current situation. New user stories are also generated, together with their tasks. After all tasks have been generated, we play a game called "planning poker" were we essentially as a group try to estimate the relative time it would take to complete each of the tasks. This is done to help figure out how many should be assigned to each task, and also to compare with previous sprints to estimate how fast the team needs to work.

Reviewer: ```suggestion
Sprint planning is done at the start of every sprint. The plan is created by the entire Scrum team working together. It should take at most 8 hours for 1-month sprints, so in our case, it should at most take 4 hours since our sprint duration is 2 weeks. During the sprint planning, we show the progress since the last sprint planning to the product owner, after which the product backlog gets updated with new requirements that build upon the current situation. New user stories are also generated, together with their tasks. After all, tasks have been generated, we play a game called "planning poker" were we essentially as a group try to estimate the relative time it would take to complete each of the tasks. This is done to help figure out how many should be assigned to each task, and also to compare with previous sprints to estimate how fast the team needs to work.
```

Author: Hvis man læser sætningen føles det komma forkert. Tror dog også jeg så grammarly foreslå det.

Hvis man sætter komma så kommer det til at blive "efter som" i stedet for "efter alle"

Reviewer: Det kan betyde begge dele

> Sprint planning is done at the start of every sprint. The plan is created by the entire Scrum team working together. It should take at most 8 hours for 1-month sprints, so in our case, it should at most take 4 hours since our sprint duration is 2 weeks. During the sprint planning, we show the progress since the last sprint planning to the product owner, after which the product backlog gets updated with new requirements that build upon the current situation. New user stories are also generated, together with their tasks. After all tasks have been generated, we play a game called "planning poker" were we essentially as a group try to estimate the relative time it would take to complete each of the tasks. This is done to help figure out how many should be assigned to each task, and also to compare with previous sprints to estimate how fast the team needs to work.

Reviewer: Jeg tror formuleringen omkring "New user stories are also generated" skal omformuleres, da de ikke bliver genereret men udarbejdet

> For each sprint, we also have a sprint goal from which we create the user stories and tasks. The sprint goal is also generated with the help of the product owner.

Reviewer: generated igen

> The sprint review is done after sprint planning. The sprint review lasts at most 4 hours for a 1-month sprint, so it should be at most 2 hours for us. In the sprint review, we take a look at all the tasks that have been completed and all of those that have not been completed. We also take a look at the items in the product backlog, and some of them might be changed or modified.

Reviewer: ```suggestion
The sprint review is done after sprint planning. The sprint review lasts at most 4 hours for a 1-month sprint, so it should be at most 2 hours in our case. In the sprint review, we take a look at all the tasks that have been completed and those that are unfinished. We also take a look at the items in the product backlog because some of them might need to be changed or modified.
```

> The sprint retrospective is done after the sprint review. The sprint review should at most last 3 hours for a 1-month sprint, so about 1.5 hours for us. In the sprint retrospective, we look back at the previous sprint to determine what went well and what went poorly and how to improve on the things that went poorly.

Reviewer: ```suggestion
The sprint retrospective is done after the sprint review. The sprint review should at most last 3 hours for a 1-month sprint, so about 1.5 hours in our case. In the sprint retrospective, we take a look at the previous sprint to determine what went well and what went poorly and how to improve on the things that went poorly.
```

> The daily Scrum is a short meeting lasting at most 15 minutes and is held daily. We have chosen to always have our Scrum meeting at 11.30. During this daily Scrum, each member of the team tells what he has been doing since the previous daily Scrum, what he will be doing until the next daily Scrum, and if there are any obstacles in his way.

Reviewer: ```suggestion
The daily Scrum is a short meeting lasting at most 15 minutes and is held daily. We have chosen to always have our Scrum meeting at 11.30. During this daily Scrum, each member of the team tells what they has been doing since the previous daily Scrum, what he will be doing till the next daily Scrum, and if there are any obstacles in his way.
```

Reviewer: Der skal måske tilføjes noget med hvad vi gør for at løse problemerne

The Scrum master is responsible for ensuring that the Scrum framework is used correctly, and to mentor others within the group about the Scrum framework.

The product owner is responsible for maximising the value of the product. The product owner is the one that manages the product backlog.

The task for the development team is to deliver increments of "done" at the end of every sprint. To determine if something is actually "done" we need to have a definition of "done". For us, something is "done" once it has been reviewed and approved by at least 2 group members.

The Scrum framework contains several events that are useful for the development process. These are: The sprint, sprint planning, sprint goal, sprint review, sprint retrospective and the daily Scrum.

The sprint is a duration of time usually less than a month and in our case just 2 weeks. During a sprint, no changes can be made that would alter the sprint goal. Within this time frame, an amount of work is "done" which should result in a new increment of the product.

> Sprint planning is done at the start of every sprint. The plan is created by the entire Scrum team working together. It should take at most 8 hours for 1-month sprints, so in our case, it should at most take 4 hours since our sprint duration is 2 weeks. During the sprint planning, we show the progress since the last sprint planning to the product owner, after which the product backlog gets updated with new requirements that build upon the current situation. New user stories are also developed, together with their tasks. After all tasks have been developed, we play a game called "planning poker" were we essentially as a group try to estimate the relative time it would take to complete each of the tasks. This is done to help figure out how many should be assigned to each task, and also to compare with previous sprints to estimate how fast the team needs to work.

Reviewer 2: ```suggestion
Sprint planning is done at the start of every sprint. The plan is created by the entire Scrum team working together. It should take at most 8 hours for 1-month sprints, so in our case, it should at most take 4 hours since our sprint duration is 2 weeks. During the sprint planning, we show the progress since the last sprint planning to the product owner, after which the product backlog gets updated with new requirements that build upon the current situation. New user stories are also developed, together with their tasks. After all tasks have been developed, we play a game called "planning poker" were we essentially as a group try to estimate the relative time in terms of story points it would take to complete each of the tasks. This is done to estimate how many story points the team can complete in a sprint (sprint velocity) and use this to select an appropriate amount user stories for a sprint.
```

Reviewer 2: Se på den den sidste sætning
